src/Microsoft.ApplicationInsights/PreciseTimestamp.cs

```diff
@@ -0,0 +1,109 @@
+﻿namespace Microsoft.ApplicationInsights
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+#if NET45 || NET46
+    using System.Diagnostics;
+    using System.Threading;
+#endif
+
+    internal class PreciseTimestamp
+    {
+        private static readonly object Lck = new object();
+        private static PreciseTimestamp instance = null;
+
+#if NET45 || NET46
+        private static TimeSync timeSync;
+        private readonly Timer syncTimeUpdater;
+
+        private PreciseTimestamp()
+        {
+            this.syncTimeUpdater = InitializeSyncTimer();
+        }
+#endif
+
+        public static PreciseTimestamp Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    lock (Lck)
+                    {
+                        if (instance == null)
+                        {
+                            instance = new PreciseTimestamp();
+                        }
+                    }
+                }
+
+                return instance;
+            }
+        }
+
+        /// <summary>
+        /// Returns high resolution (1 DateTime tick) current UTC DateTime. 
+        /// </summary>
+        [SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic", Justification = "Enforcing instance fields initialization.")]
+        internal DateTimeOffset GetUtcNow()
```

Comment:
Is there a reason to use `DateTimeOffset` over simply `DateTime`? 
DateTimeOffset is a larger struct and it seems like we really don't need to know the time-zone -- since it's always UTC.

Reply:
because AppInsights uses offset everywhere